tests/integration/test_types_integration.py

```diff
@@ -142,6 +145,171 @@ def test_json(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=date(2001, 8, 22)) \
+        .add_field(sql="DATE '0001-01-01'", python=date(1, 1, 1)) \
+        .add_field(sql="DATE '1582-10-04'", python=date(1582, 10, 4)) \
+        .add_field(sql="DATE '1582-10-05'", python=date(1582, 10, 5)) \
+        .add_field(sql="DATE '1582-10-14'", python=date(1582, 10, 14)) \
+        .execute()
+
+
+def test_time(trino_connection):
+    time_0 = time(1, 23, 45)
+    time_3 = time(1, 23, 45, 123000)
+    # Because of `PARAMETRIC_DATETIME` not being set, values are rounded to milliseconds
+    time_6 = time(1, 23, 45, 123000)
+    time_round_down = time(1, 23, 45, 123000)
+    time_round_up = time(1, 23, 45, 124000)
+
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS TIME)", python=None) \
+        .add_field(sql="CAST(null AS TIME(0))", python=None) \
+        .add_field(sql="CAST(null AS TIME(3))", python=None) \
+        .add_field(sql="CAST(null AS TIME(6))", python=None) \
+        .add_field(sql="CAST(null AS TIME(9))", python=None) \
+        .add_field(sql="CAST(null AS TIME(12))", python=None) \
+        .add_field(sql="CAST('01:23:45' AS TIME(0))", python=time_0) \
+        .add_field(sql="TIME '01:23:45.123'", python=time_3) \
+        .add_field(sql="CAST('01:23:45.123' AS TIME(3))", python=time_3) \
```

Comment:
What we want to test is type mapping between Python client and Trino server, not literal and cast in Trino server. I suppose we can remove either one. If we still want to test both, `test_date` misses the cast coverage.

Reply:
It is indeed useless. I have removed the casts also we should not test precision of null values so I kept only one null test for each data type.